<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Preferences</class>
 <widget class="QDialog" name="Preferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>631</width>
    <height>403</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="editorTab">
        <attribute name="title">
         <string>Editor</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="fontLabel">
             <property name="text">
              <string>Font:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFontComboBox" name="fontComboBox">
             <property name="fontFilters">
              <set>QFontComboBox::MonospacedFonts</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="sizeComboBox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>64</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="editable">
              <bool>true</bool>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>243</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCheckBox" name="darkThemeCheckBox">
           <property name="text">
            <string>Use dark theme (requires restart)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="highlightLineCheckbox">
           <property name="text">
            <string>Highlight current line</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="tooltipsCheckBox">
           <property name="text">
            <string>Show tooltips for built in modules</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="modelColorsTab">
        <attribute name="title">
         <string>Model Colors</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="2">
          <widget class="QLabel" name="markedEdgeColorLabel">
           <property name="text">
            <string>Marked Edge Color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QToolButton" name="markedFacetColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QToolButton" name="vertexColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QFrame" name="markedVertexColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QToolButton" name="edgeColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QDoubleSpinBox" name="vertexSizeSpinBox"/>
         </item>
         <item row="7" column="0">
          <widget class="QDoubleSpinBox" name="edgeSizeSpinBox"/>
         </item>
         <item row="2" column="1">
          <widget class="QToolButton" name="markedEdgeColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QToolButton" name="markedVertexColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="edgeColorLabel">
           <property name="text">
            <string>Edge Color</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QToolButton" name="facetColorToolButton">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QFrame" name="markedEdgeColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="markedFacetColorLabel">
           <property name="text">
            <string>Marked Facet Color</string>
           </property>
          </widget>
         </item>
         <item row="7" column="2">
          <widget class="QLabel" name="edgeSizeLabel">
           <property name="text">
            <string>Edge Size</string>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QLabel" name="vertexColorLabel">
           <property name="text">
            <string>Vertex Color</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QFrame" name="facetColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="6" column="2">
          <widget class="QLabel" name="vertexSizeLabel">
           <property name="text">
            <string>Vertex Size</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="facetColorLabel">
           <property name="text">
            <string>Facet Color</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QFrame" name="markedFacetColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QFrame" name="edgeColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QFrame" name="vertexColorFrame">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="frameShape">
            <enum>QFrame::StyledPanel</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QLabel" name="markedVertexColorLabel">
           <property name="text">
            <string>Marked Vertex Color</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="printBedTab">
        <attribute name="title">
         <string>Print Bed</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QHBoxLayout" name="dimensionsLayout">
           <item>
            <widget class="QGroupBox" name="volumeGroupBox">
             <property name="title">
              <string>Volume</string>
             </property>
             <layout class="QFormLayout" name="formLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="widthLabel">
                <property name="text">
                 <string>Width:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="lengthLabel">
                <property name="text">
                 <string>Length:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="heightLabel">
                <property name="text">
                 <string>Height:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="widthSpinBox">
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="lengthSpinBox">
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSpinBox" name="heightSpinBox">
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="originGroupBox">
             <property name="title">
              <string>Origin</string>
             </property>
             <layout class="QFormLayout" name="formLayout_2">
              <item row="0" column="0">
               <widget class="QLabel" name="Xlabel">
                <property name="text">
                 <string>X:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="Ylabel">
                <property name="text">
                 <string>Y:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="YspinBox">
                <property name="minimum">
                 <number>-10000</number>
                </property>
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSpinBox" name="XspinBox">
                <property name="minimum">
                 <number>-10000</number>
                </property>
                <property name="maximum">
                 <number>10000</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QCheckBox" name="translateCheckBox">
           <property name="text">
            <string>Translate output</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="printBedVerticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="appearanceLayout" stretch="0,0">
           <property name="sizeConstraint">
            <enum>QLayout::SetDefaultConstraint</enum>
           </property>
           <item>
            <widget class="QLabel" name="appearanceLabel">
             <property name="text">
              <string>Appearance:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="appearanceComboBox">
             <item>
              <property name="text">
               <string>MK42</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>MK2</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="outputOptions">
        <attribute name="title">
         <string>Output</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QGroupBox" name="precisionGroupBox">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="title">
            <string>Output Precision</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="1" column="1">
             <widget class="QSpinBox" name="placesSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>31</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QRadioButton" name="singleRadio">
              <property name="text">
               <string>Single</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="placesLabel">
              <property name="text">
               <string>Decimal Places:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="bitsLabel">
              <property name="text">
               <string>Significand Bits:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QRadioButton" name="doubleRadio">
              <property name="text">
               <string>Double</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QRadioButton" name="customRadio">
              <property name="text">
               <string>Custom</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QSpinBox" name="bitsSpinBox">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>3</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="roundingGroupBox">
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="title">
            <string>Function Rounding</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QRadioButton" name="decimalRoundingRadio">
              <property name="text">
               <string>Decimal</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="base2RoundingRadio">
              <property name="text">
               <string>Base 2</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="noRoundingRadio">
              <property name="text">
               <string>None</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="formatGroupBox">
           <property name="title">
            <string>Format</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QRadioButton" name="scientificRadio">
                <property name="text">
                 <string>Scientific</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="rationalRadio">
                <property name="text">
                 <string>Rational</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="fixedRadio">
                <property name="text">
                 <string>Fixed</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QLabel" name="exampleLabel">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="text">
               <string>0.0</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBox">
           <property name="text">
            <string>Always save files before compile and render</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="externalOptions">
        <attribute name="title">
         <string>External</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="launchCommandLabel">
             <property name="text">
              <string>CAM launch command</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="launchCommandLineEdit"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="fileTypeLabel">
             <property name="text">
              <string>File type</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="_3mfRadioButton">
             <property name="text">
              <string>3mf</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="AMFRadioButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>AMF</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="exampleCommandLabel">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QLabel" name="processingScriptlabel">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>CAM Processing script</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="processingScriptlineEdit">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QCheckBox" name="showGCODEButtonCheckbox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Show GCODE generation button</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>fontComboBox</tabstop>
  <tabstop>sizeComboBox</tabstop>
  <tabstop>highlightLineCheckbox</tabstop>
  <tabstop>tooltipsCheckBox</tabstop>
  <tabstop>markedFacetColorToolButton</tabstop>
  <tabstop>facetColorToolButton</tabstop>
  <tabstop>markedEdgeColorToolButton</tabstop>
  <tabstop>edgeColorToolButton</tabstop>
  <tabstop>markedVertexColorToolButton</tabstop>
  <tabstop>vertexColorToolButton</tabstop>
  <tabstop>vertexSizeSpinBox</tabstop>
  <tabstop>edgeSizeSpinBox</tabstop>
  <tabstop>widthSpinBox</tabstop>
  <tabstop>lengthSpinBox</tabstop>
  <tabstop>heightSpinBox</tabstop>
  <tabstop>XspinBox</tabstop>
  <tabstop>YspinBox</tabstop>
  <tabstop>appearanceComboBox</tabstop>
  <tabstop>singleRadio</tabstop>
  <tabstop>doubleRadio</tabstop>
  <tabstop>customRadio</tabstop>
  <tabstop>placesSpinBox</tabstop>
  <tabstop>bitsSpinBox</tabstop>
  <tabstop>decimalRoundingRadio</tabstop>
  <tabstop>base2RoundingRadio</tabstop>
  <tabstop>noRoundingRadio</tabstop>
  <tabstop>scientificRadio</tabstop>
  <tabstop>rationalRadio</tabstop>
  <tabstop>fixedRadio</tabstop>
  <tabstop>checkBox</tabstop>
  <tabstop>launchCommandLineEdit</tabstop>
  <tabstop>_3mfRadioButton</tabstop>
  <tabstop>AMFRadioButton</tabstop>
  <tabstop>processingScriptlineEdit</tabstop>
  <tabstop>showGCODEButtonCheckbox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Preferences</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>620</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Preferences</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>620</x>
     <y>340</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
